File: tkpddiscovery/src/main/java/com/tokopedia/discovery/newdynamicfilter/helper/FilterDbHelper.java
package com.tokopedia.discovery.newdynamicfilter.helper;

import android.content.Context;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.tokopedia.discovery.common.data.Filter;
import com.tokopedia.discovery.common.data.Option;

import java.lang.reflect.Type;
import java.util.List;

/**
 * Created by henrypriyono on 11/27/17.
 */

public class FilterDbHelper {
    public static void storeLocationFilterOptions(Context context, List<Option> optionList) {

        Type listType = new TypeToken<List<Option>>() {}.getType();
        Gson gson = new Gson();
        String optionData = gson.toJson(optionList, listType);

        DynamicFilterDbManager.store(context, Filter.TEMPLATE_NAME_LOCATION, optionData);
    }

    public static List<Option> loadLocationFilterOptions(Context context) {
        String data = DynamicFilterDbManager.getFilterData(context, Filter.TEMPLATE_NAME_LOCATION);
        Type listType = new TypeToken<List<Option>>() {}.getType();
        return new Gson().fromJson(data, listType);
    }
}

File: features/discovery/filter/src/main/java/com/tokopedia/filter/newdynamicfilter/helper/FilterDbHelper.kt
package com.tokopedia.filter.newdynamicfilter.helper

import android.content.Context

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.tokopedia.filter.common.data.Filter
import com.tokopedia.filter.common.data.Option

/**
 * Created by henrypriyono on 11/27/17.
 */

object FilterDbHelper {
    fun storeLocationFilterOptions(context: Context, optionList: List<Option>) {

        val listType = object : TypeToken<List<Option>>() {}.type
        val gson = Gson()
        val optionData = gson.toJson(optionList, listType)

        DynamicFilterDbManager.store(context, Filter.TEMPLATE_NAME_LOCATION, optionData)
    }

    fun loadLocationFilterOptions(context: Context): List<Option> {
        val data = DynamicFilterDbManager.getFilterData(context, Filter.TEMPLATE_NAME_LOCATION)
        val listType = object : TypeToken<List<Option>>() {}.type
        return Gson().fromJson(data, listType)
    }
}
